bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValueImpl.java

```diff
@@ -808,8 +813,9 @@ protected void fillValues(int index) {
             case TypeTags.BOOLEAN_TAG:
                 return;
             default:
-                Arrays.fill(refValues, size, index, elementType.getZeroValue());
-
+                if (arrayType.hasFillerValue()) {
```

Comment:
Do we need to check this here?

Isn't `fillValues` always called after `fillerValueCheck`?

Reply:
This is not true for code coming from desugar.
As an example.

```
     AbstractPersonObject[5][] y = [];
     AbstractPersonObject e1 = new Employee("John", "Doe");
     y[0] = [e1];
     AbstractPersonObject[][5] r = [];
     r[0] = [e1, e1, e1, e1, e1];
```
```r[0] = [e1, e1, e1, e1, e1]``` expands by first creating a sealed array. Then adding values.
```r[0] => 
AbstractPersonObject[5] = [] 
```
 This is not allowed we cannot fill this by 5 AbstractPersonObject s. If user write this code it will give a compiler error . But since this is tool generated implementation we cannot error out this. 
```fillerValueCheck``` error out only if there is a need for filling. Consecutive inserts should not fail too which is guranteed by ```index > size``` condition. Since this array size is already 5 as generated by desugar ```fillerValueCheck``` at runtime will not give an error.